internal/admin/admin.go

```diff
@@ -544,35 +546,50 @@ func (adm *Admin) updateLogStream(ctx context.Context, lsid types.LogStreamID, p
 
 	clusmeta, err := adm.mrmgr.ClusterMetadataView().ClusterMetadata(ctx)
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Unavailable, "update log stream: %s", err.Error())
 	}
 
 	oldLSDesc, err := clusmeta.MustHaveLogStream(lsid)
-	if err != nil {
-		return nil, err
+	if err != nil || oldLSDesc.Status.Deleted() {
+		return nil, status.Errorf(codes.NotFound, "update log stream: no such log stream %d", lsid)
 	}
 
-	status := oldLSDesc.GetStatus()
-	if status.Running() || status.Deleted() {
-		return nil, errors.Errorf("invalid log stream status: %s", status)
+	if oldLSDesc.Status.Running() {
+		return nil, status.Errorf(codes.FailedPrecondition, "update log stream: invalid log stream status %s", oldLSDesc.Status)
```

Comment:
I plan to treat `FailedPrecondition` as a not-executable state. 
Does the running state mean that the UpdateLogStream has already been applied?

Reply:
To update replicas in a log stream, it should be sealed first.
I am not sure what `not-executable state` means, however it is sure that `codes.FailedPrecondition` means that the log stream cannot be updated with just a simple retry.